### DISCUSSION TITLE
Is there a law to get the same experience as a spa project?

### DISCUSSION THREAD

**Discussion by cong1223** [vercel/next.js#47134] [Help] [answered]
### Summary

I think it is necessary for next to open a configuration to tell it which pages are, so that he will only focus on packaging pages when packaging. The problem now is that as long as I put the component under the page, it will be packaged into an html file, which will affect the packaging efficiency, Another problem is that the index page cannot be read from the index folder.
My ideal directory structure is

-- pages
[][]-- index
[][][][] -- components
[][][][][][]        -- Header
      [][][][][][][][]             -- index.tsx
      [][][][]   -- index.tsx

**Comment by icyJoseph** [comment, marked as answer]
I'd recommend going through the docs: https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions#including-non-page-files-in-the-pages-directory

That being said, personal opinion here, file structures that blend pages and components, often break down as soon as pages start to depend on components found in other pages. Quick recipe for a big ball of mud architecture.
